File: Python/Algorithms/1302.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:

    def deepestLeavesSum(self, root):
        l_sum = depth = 0
        def traverse(node, d=0):
            nonlocal depth, l_sum
            val, left, right = node.val, node.left, node.right

            if not left and not right:
                if d > depth:
                    depth = d
                    l_sum = val
                elif d == depth:
                    l_sum = l_sum + val
            else:
                if left:
                    traverse(left, d + 1)
                if right:
                    traverse(right, d + 1)
        traverse(root)
        return l_sum

File: Javascript/Algorithms/1302.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
let deepestLeavesSum = root => {
    let l_sum = 0, depth = 0;
    let traverse = (node, d) => {
        let {val, left, right} = node;

        if (!left && !right){
            if (d > depth){
                depth = d;
                l_sum = val;
            } else if (d == depth)
                l_sum += val;
        } else {
            if (left)
                traverse(left, d + 1);
            if (right)
                traverse(right, d + 1);
        }
    }
    traverse(root, 0)
    return l_sum;    
};
